src/prof.c

```diff
@@ -106,6 +108,9 @@ static uint64_t		prof_dump_mseq;
 static uint64_t		prof_dump_useq;
 
 malloc_mutex_t	prof_dump_mtx;
+static char	prof_dump_prefix[PROF_DUMP_PREFIX_LEN] = {0};
+static bool	prof_dump_prefix_override = false;
+static malloc_mutex_t	prof_dump_prefix_mtx;
```

Comment:
I would think using `prof_dump_mtx` as the guard should be fine? cc: @yinan1048576 for sanity check.

Reply:
Hmm. If we want to reuse an existing mutex, I'd think `prof_dump_seq_mtx` is functionally closer. Both the prefix and the sequence numbers are for the file names. (We can probably change the mutex's name to be something better.)

Reply:
I actually think `prof_dump_seq_mtx` is more similar, but I didn't think of using it, because I just ignored I can actually use a coarse-grained lock for multiple shared values.  Maybe we can rename it to be something like `prof_dump_filename_mtx` first, then use that in this feature? @yinan1048576

Reply:
`prof_dump_filename_mtx` sounds good to me. @interwq thoughts?

I remember `prof_dump_seq_mtx` might have already been used for multiple values, e.g. those different kinds of sequence numbers, so adding one more doesn't seem like crazy.

I think another benefit of consolidating `prof_dump_seq_mtx` with `prof_dump_prefix_mtx` is that you don't need to always try to acquire both of them when getting the file name. Feels cleaner and safer.

Reply:
SGTM!